BritneyS/dnd-graphql
# Initial setup

## What does this PR do :question:
This PR contains the initial setup for the Node Apollo GraphQL server for this project.


## How to test it :microscope:
1. Check out this branch, and navigate to this project's root directory in the terminal
2. Install dependencies by running `npm install`
3. In the project's root directory, add a file called `.env`, and include the line `ENV_PORT=8000` in the file
4. Run `npm start`
5. In the terminal, nodemon should execute and the strings `Hello Node.js project.` and `Apollo Server on http://localhost:8000/graphql` should be printed in the console (see screenshot)
6. In the browser, navigate to `http://localhost:8000/graphql`, and note that the GraphQL Playground environment should be running (see screenshot)
7. In the query window of the playground environment, query the following: 
```
{
  foo {
      name
  }
}
```
Run the query, and note the expected response:

```
{
  "data": {
    "foo": {
      "name": "Britney Smith"
    }
  }
}
```
(see screenshot)

## Any background info you would like to include :white_check_mark:
N/A


## Screenshots (if applicable) :camera:
<img width="450" alt="Screen Shot 2019-05-10 at 11 05 58 AM" src="https://user-images.githubusercontent.com/8409475/57537187-bf25a900-7313-11e9-88ff-ce5af2dec68e.png">
<img width="802" alt="Screen Shot 2019-05-10 at 11 06 36 AM" src="https://user-images.githubusercontent.com/8409475/57537194-c2b93000-7313-11e9-85fb-2b4acf976f3f.png">
